### PR TITLE
[zh-cn]: add translation for `Navigator.share()`

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Related issues and pull requests

part of the #19537

### CODE REVIEW

File: files/zh-cn/web/api/navigator/share/index.md [hunk 1]
```diff
@@ -1,66 +1,217 @@
...
-const sharePromise = window.navigator.share(data);
+```js-nolint
+navigator.share(data)
 ```
 
 ### 参数
 
-- _data_
+- `data` {{optional_inline}}
+  - : 包含要共享的数据的对象。通常，如果此调用返回 `true`，则具有相同属性的对象将传递给 {{domxref("navigator.share()")}}。用户代理未知的属性将被忽略；共享数据仅根据用户代理理解的属性进行评估。所有属性都是可选的，但必须至少指定一个已知的数据属性。可能的值为：
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/navigator/share/index.md:25]:
```suggestion
  - : 包含要共享的数据的对象。用户代理未知的属性将被忽略；共享数据仅根据用户代理理解的属性进行评估。所有属性都是可选的，但必须至少指定一个已知的数据属性。可能的值为：
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 2]
```diff
@@ -1,66 +1,217 @@
...
+一个兑现为 `undefined` 的 {{jsxref("Promise")}}，或者被下面给出的[异常](#exceptions)之一拒绝。
+
+### 异常
+
+{{jsxref("Promise")}} 可能会被拒绝并带有以下 `DOMException` 值之一：
+
+- `InvalidStateError` {{domxref("DOMException")}}
+  - : 文档未完全激活，或正在进行其他共享操作。
+- `NotAllowedError` {{domxref("DOMException")}}
+  - : `web-share` [权限策略](/zh-CN/docs/Web/HTTP/Permissions_Policy)已用于阻止使用此功能，或当前文档没有{{Glossary("transient activate", "瞬态激活")}}，或者出于安全考虑，文件共享被阻止。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/navigator/share/index.md:46]:
```suggestion
  - : `web-share` [权限策略](/zh-CN/docs/Web/HTTP/Permissions_Policy)已用于阻止使用此功能，或当前文档没有{{Glossary("transient activation", "瞬态激活")}}，或者出于安全考虑，文件共享被阻止。
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 3]
```diff
@@ -1,66 +1,217 @@
...
-const sharePromise = window.navigator.share(data);
+```js-nolint
+navigator.share(data)
 ```
 
 ### 参数
 
-- _data_
+- `data` {{optional_inline}}
+  - : 包含要共享的数据的对象。用户代理未知的属性将被忽略；共享数据仅根据用户代理理解的属性进行评估。所有属性都是可选的，但必须至少指定一个已知的数据属性。可能的值为：
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:25]:
```suggestion
  - : 包含要共享的数据的对象。

    用户代理未知的属性将被忽略；共享数据仅根据用户代理可识别的属性进行评估。所有属性都是可选的，但必须至少指定一个已知的数据属性。

    可能的值有：
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 4]
```diff
@@ -1,66 +1,217 @@
...
+    - `url` {{optional_inline}}
+      - : 表示要共享的 URL 的字符串。
+    - `text` {{optional_inline}}
+      - : 表示要共享的文本的字符串。
+    - `title` {{optional_inline}}
+      - : 表示要共享的标题的字符串。
+    - `files` {{optional_inline}}
+      - : 表示要共享的文件的 {{domxref("File")}} 对象数组。有关可共享的文件类型，请参阅[下文](#shareable_file_types)。
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:33]:
```suggestion
      - : 表示要共享的文件的 {{domxref("File")}} 对象数组。有关可共享的文件类型，请参阅[下文](#可分享的文件类型)。
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 5]
```diff
@@ -1,66 +1,217 @@
...
 
-  - : 包含要共享的数据的对象。必须至少指定以下字段之一。可用选项包括：
+### 返回值
 
-    - `url`: 要共享的 URL（ {{domxref("USVString")}} ）
-    - `text`: 要共享的文本（ {{domxref("USVString")}} ）
-    - `title`: 要共享的标题（ {{domxref("USVString")}}）
-    - `files`: 要共享的文件（“FrozenArray”）
+一个兑现为 `undefined` 的 {{jsxref("Promise")}}，或者被下面给出的[异常](#exceptions)之一拒绝。
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:37]:
```suggestion
一个兑现为 `undefined` 的 {{jsxref("Promise")}}，或者被下面给出的[异常](#异常)之一拒绝。
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 6]
```diff
@@ -1,66 +1,217 @@
...
+
+### 异常
+
+{{jsxref("Promise")}} 可能会被拒绝并带有以下 `DOMException` 值之一：
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:41]:
```suggestion
{{jsxref("Promise")}} 可能会以下列 `DOMException` 值之一被拒绝：
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 7]
```diff
@@ -1,66 +1,217 @@
...
+
+- `InvalidStateError` {{domxref("DOMException")}}
+  - : 文档未完全激活，或正在进行其他共享操作。
+- `NotAllowedError` {{domxref("DOMException")}}
+  - : `web-share` [权限策略](/zh-CN/docs/Web/HTTP/Permissions_Policy)已用于阻止使用此功能，或当前文档没有{{Glossary("transient activation", "瞬态激活")}}，或者出于安全考虑，文件共享被阻止。
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:46]:
```suggestion
  - : `web-share` [权限策略](/zh-CN/docs/Web/HTTP/Permissions_Policy)已用于阻止使用此特性，或当前文档没有{{Glossary("transient activation", "瞬态激活")}}，或者出于安全考虑，文件共享被阻止。
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 8]
```diff
@@ -1,66 +1,217 @@
...
+  - `.text` - `text/plain`
+  - `.txt` - `text/plain`
+- 视频
+  - `.m4v` - `video/mp4`
+  - `.mp4` - `video/mp4`
+  - `.mpeg` - `video/mpeg`
+  - `.mpg` - `video/mpeg`
+  - `.ogm` - `video/ogg`
+  - `.ogv` - `video/ogg`
+  - `.webm` - `video/webm`
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:107]: nits, replace English punctuations

```suggestion
  - `.pdf`——`application/pdf`
- 音频
  - `.flac`——`audio/flac`
  - `.m4a`——`audio/x-m4a`
  - `.mp3`——`audio/mpeg`（也接受 `audio/mp3`）
  - `.oga`——`audio/ogg`
  - `.ogg`——`audio/ogg`
  - `.opus`——`audio/ogg`
  - `.wav`——`audio/wav`
  - `.weba`——`audio/webm`
- 图像
  - `.avif`——`image/avif`
  - `.bmp`——`image/bmp`
  - `.gif`——`image/gif`
  - `.ico`——`image/x-icon`
  - `.jfif`——`image/jpeg`
  - `.jpeg`——`image/jpeg`
  - `.jpg`——`image/jpeg`
  - `.pjp`——`image/jpeg`
  - `.pjpeg`——`image/jpeg`
  - `.png`——`image/png`
  - `.svg`——`image/svg+xml`
  - `.svgz`——`image/svg+xml`
  - `.tif`——`image/tiff`
  - `.tiff`——`image/tiff`
  - `.webp`——`image/webp`
  - `.xbm`——`image/x-xbitmap`
- 文本
  - `.css`——`text/css`
  - `.csv`——`text/csv`
  - `.ehtml`——`text/html`
  - `.htm`——`text/html`
  - `.html`——`text/html`
  - `.shtm`——`text/html`
  - `.shtml`——`text/html`
  - `.text`——`text/plain`
  - `.txt`——`text/plain`
- 视频
  - `.m4v`——`video/mp4`
  - `.mp4`——`video/mp4`
  - `.mpeg`——`video/mpeg`
  - `.mpg`——`video/mpeg`
  - `.ogm`——`video/ogg`
  - `.ogv`——`video/ogg`
  - `.webm`——`video/webm`
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 9]
```diff
@@ -1,66 +1,217 @@
...
+
+## 安全
+
+此方法要求当前文档具有 [web-share](/zh-CN/docs/Web/HTTP/Headers/Permissions-Policy/web-share) 权限策略和{{Glossary("transient activation", "瞬态激活")}}。（它必须由 UI 事件触发，例如单击按钮，并且不能由脚本在任意点启动。）此外，该方法必须指定本机实现的支持共享的有效数据。
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:111]:
```suggestion
此方法要求当前文档具有 [web-share](/zh-CN/docs/Web/HTTP/Headers/Permissions-Policy/web-share) 权限策略和{{Glossary("transient activation", "瞬态激活")}}状态。（它必须由 UI 事件触发，例如单击按钮，并且不能由脚本在任意点启动。）此外，该方法必须指定本机实现的支持共享的有效数据。
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 10]
```diff
@@ -1,66 +1,217 @@
...
 
-### 返回值
+## 示例
 
-该方法将会返回一个{{jsxref("Promise")}}。一旦用户完成分享，这个 promise 将会接受。如果指定的共享数据格式不正确，promise 将会立即拒绝；如果用户取消了分享，promise 也会拒绝。
+### 分享一个 URL
 
-例如，在 Android 的 Chrome 上，将在用户选择要共享的应用程序后将会解析共享的内容。
+下面的示例显示了单击按钮调用 Web 共享 API 来共享 MDN 的 URL。这是取自我们的[网络共享测试](https://mdn.github.io/dom-examples/web-share/)（[查看源代码](https://github.com/mdn/dom-examples/blob/main/web-share/index.html)）。
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:117]:
```suggestion
下面的示例显示了单击按钮调用 Web 共享 API 来共享 MDN 的 URL。这是取自我们的 [Web 共享测试](https://mdn.github.io/dom-examples/web-share/)（[查看源代码](https://github.com/mdn/dom-examples/blob/main/web-share/index.html)）。
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 11]
```diff
@@ -1,66 +1,217 @@
...
+
+document.getElementById("share").addEventListener("click", async () => {
+  const files = input.files;
+
+  if (files.length === 0) {
+    output.textContent = "未选择任何文件。";
+    return;
+  }
+
+  // 功能检测 navigator.canShare() 也意味着 navigator.share() 的功能相同
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:188]:
```suggestion
  // 检测 navigator.canShare() 特性和检测 navigator.share() 特性是一样的
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 12]
```diff
@@ -1,66 +1,217 @@
...
+  }
+
+  if (navigator.canShare({ files })) {
+    try {
+      await navigator.share({
+        files,
+        title: "图像",
+        text: "美丽的图像",
+      });
+      output.textContent = "分享了！";
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:201]:
```suggestion
      output.textContent = "分享成功！";
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 13]
```diff
@@ -72,3 +223,4 @@ if (navigator.canShare && navigator.canShare({ files: filesArray })) {
 ## 参见
 
 - {{domxref("navigator.canShare()")}}
+- <https://wpt.live/web-share/>（网络平台测试）
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:226]:
```suggestion
- <https://wpt.live/web-share/>（Web 平台测试）
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 14]
```diff
@@ -1,66 +1,223 @@
...
+一个兑现为 `undefined` 的 {{jsxref("Promise")}}，或者被下面给出的[异常](#异常)之一拒绝。
+
+### 异常
+
+{{jsxref("Promise")}} 可能会以下列 `DOMException` 值之一被拒绝：
+
+- `InvalidStateError` {{domxref("DOMException")}}
+  - : 文档未完全激活，或正在进行其他共享操作。
+- `NotAllowedError` {{domxref("DOMException")}}
+  - : `web-share` [权限策略](/zh-CN/docs/Web/HTTP/Permissions_Policy)已用于阻止使用此特性，或当前文档没有{{Glossary("transient activation", "瞬态激活")}}，或者出于安全考虑，文件共享被阻止。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/navigator/share/index.md:52]:
```suggestion
  - : `web-share` [权限策略](/zh-CN/docs/Web/HTTP/Permissions_Policy)已用于阻止使用此特性，或当前文档没有{{Glossary("transient activation", "瞬态激活")}}状态，或者出于安全考虑，文件共享被阻止。
```

File: files/zh-cn/web/api/navigator/share/index.md [hunk 15]
```diff
@@ -1,66 +1,223 @@
...
+一个兑现为 `undefined` 的 {{jsxref("Promise")}}，或者被下面给出的[异常](#异常)之一拒绝。
+
+### 异常
+
+{{jsxref("Promise")}} 可能会以下列 `DOMException` 值之一被拒绝：
+
+- `InvalidStateError` {{domxref("DOMException")}}
+  - : 文档未完全激活，或正在进行其他共享操作。
+- `NotAllowedError` {{domxref("DOMException")}}
+  - : `web-share` [权限策略](/zh-CN/docs/Web/HTTP/Permissions_Policy)已用于阻止使用此特性，或当前文档没有{{Glossary("transient activation", "瞬态激活")}}状态，或者出于安全考虑，文件共享被阻止。
```
Comment (yin1999) [inline files/zh-cn/web/api/navigator/share/index.md:52]:
```suggestion
  - : `web-share` [权限策略](/zh-CN/docs/Web/HTTP/Permissions_Policy)已用于阻止使用此特性，或当前文档不处于{{Glossary("transient activation", "瞬态激活")}}状态，或者出于安全考虑，文件共享被阻止。
```

**Review by yin1999 — APPROVED** [review]
:+1:

**Pull request merged**